tardis/widgets/kromer_plot.py

```diff
@@ -0,0 +1,1201 @@
+        )
+        self.modeled_spectrum_luminosity = (
+            self.data[packets_mode].spectrum_luminosity_density_lambda[
+                self.packet_wvl_range_mask
+            ]
+            / self.lum_to_flux
+        )
+
+    def _calculate_emission_luminosities(self, packets_mode, packet_wvl_range):
+        """
+        Calculate luminosities for the emission part of Kromer plot.
+
+        Parameters
+        ----------
+        packets_mode : {'virtual', 'real'}
+            Mode of packets to be considered, either real or virtual
+        packet_wvl_range : astropy.Quantity
+            Wavelength range to restrict the analysis of escaped packets. It
+            should be a quantity having units of Angstrom, containing two
+            values - lower lambda and upper lambda i.e.
+            [lower_lambda, upper_lambda] * u.AA
+
+        Returns
+        -------
+        luminosities_df : pd.DataFrame
+            Dataframe containing luminosities contributed by no interaction,
+            only e-scattering and emission with each element present
+        elements_present: np.array
+            Atomic numbers of the elements with which packets of specified
+            wavelength range interacted
+        """
+        # Calculate masks to be applied on packets data based on packet_wvl_range
+        if packet_wvl_range is None:
+            self.packet_nu_range_mask = np.ones(
+                self.data[packets_mode].packets_df.shape[0], dtype=bool
+            )
+            self.packet_nu_line_range_mask = np.ones(
+                self.data[packets_mode].packets_df_line_interaction.shape[0],
+                dtype=bool,
+            )
+        else:
+            packet_nu_range = packet_wvl_range.to("Hz", u.spectral())
+            # print(packet_nu_range)
+            self.packet_nu_range_mask = (
+                self.data[packets_mode].packets_df["nus"] < packet_nu_range[0]
+            ) & (self.data[packets_mode].packets_df["nus"] > packet_nu_range[1])
+            self.packet_nu_line_range_mask = (
+                self.data[packets_mode].packets_df_line_interaction["nus"]
+                < packet_nu_range[0]
+            ) & (
+                self.data[packets_mode].packets_df_line_interaction["nus"]
+                > packet_nu_range[1]
+            )
+
+        # Histogram weights are packet luminosities or flux
+        weights = (
+            self.data[packets_mode].packets_df["energies"][
+                self.packet_nu_range_mask
+            ]
+            / self.lum_to_flux
+        ) / self.data[packets_mode].time_of_simulation
+
+        luminosities_df = pd.DataFrame(index=self.plot_wavelength)
+
+        # Contribution of packets which experienced no interaction ------------
+        # Mask to select packets with no interaction
+        mask_noint = (
+            self.data[packets_mode].packets_df["last_interaction_type"][
+                self.packet_nu_range_mask
+            ]
+            == -1
```

Comment:
Hard coded numbers fine for now, but will be changed in the future.

Reply:
Created an issue #1396